pkg/asset/installconfig/gcp/validation.go

```diff
@@ -121,15 +119,21 @@ func ValidatePreExitingPublicDNS(client API, ic *types.InstallConfig) error {
 		return nil
 	}
 
+	// skip validation when Public Managed Zone is set to "None"
+	if strings.EqualFold(ic.GCP.PublicManagedZone, "None") {
+		return nil
+	}
+
 	record := fmt.Sprintf("api.%s.", strings.TrimSuffix(ic.ClusterDomain(), "."))
 
 	zone, err := client.GetPublicDNSZone(context.TODO(), ic.Platform.GCP.ProjectID, ic.BaseDomain)
 	if err != nil {
-		var gErr *googleapi.Error
-		if errors.As(err, &gErr) {
-			if gErr.Code == http.StatusNotFound {
-				return field.NotFound(field.NewPath("baseDomain"), fmt.Sprintf("DNS Zone (%s/%s)", ic.Platform.GCP.ProjectID, ic.BaseDomain))
-			}
+		if IsNotFound(err) {
+			return field.NotFound(field.NewPath("baseDomain"), fmt.Sprintf("DNS Zone (%s/%s)", ic.Platform.GCP.ProjectID, ic.BaseDomain))
+		}
+		if IsForbidden(err) {
+			logrus.Warn("Permission denied. Unable to fetch public dns zone information: %s", ic.BaseDomain)
+			return nil
```

Comment:
I think we want to return an error here. When we are planning on adding/removing records from a zone, we should at least have permissions to view the zone itself, right? So, if we don't have that, we should fail early.